jerry-core/ecma/builtin-objects/ecma-builtin-object.c

```diff
@@ -790,7 +795,12 @@ ecma_builtin_object_object_define_properties (ecma_object_t *obj_p, /**< routine
   for (uint32_t i = 0; i < prop_names_p->item_count; i++)
   {
     /* 5.a */
-    ecma_value_t desc_obj = ecma_op_object_get (props_p, ecma_get_string_from_value (buffer_p[i]));
+#if ENABLED (JERRY_ESNEXT)
+    ecma_string_t *prop_name_p = ecma_get_prop_name_from_value (buffer_p[i]);
```

Comment:
We can use the "get_prop_name_from" variant in every no need to have it guarded. In ES5.1 case it will only work with strings just like the "get_string_from_value".

Reply:
Done.